### PR TITLE
testing rgw using curl and testing transfer-encoding chunked with put object instead of content length

### PR DESCRIPTION
**Pull request by hmaheswa**
This PR is to add support for testing rgw features using curl and also to automate testing of customer defect getting "MissingContentLength" error while uploading object using "Transfer-Encoding:chunked" header instead of content length through curl.
Seeing "SignatureDoesNotMatch" error while performing versioning, lc, notifications, policy... operations through curl. refer [BZ2184756](https://bugzilla.redhat.com/show_bug.cgi?id=2184756) for more details. Will automate testing those features through curl once the issue in the bz is solved.

bugzilla: [BZ2152801](https://bugzilla.redhat.com/show_bug.cgi?id=2152801)
polarion id: [CEPH-83575572](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83575572)

pass logs: http://magna002.ceph.redhat.com/ceph-qe-logs/HemanthSai/curl_missing_content_len_automation/

updated pass log:http://magna002.ceph.redhat.com/ceph-qe-logs/HemanthSai/curl_missing_content_len_automation/test_rgw_using_curl_updated.console.log

### CODE REVIEW

**Review by mkasturi18 — COMMENTED** [review]
Changes looks good to me.

**Review by anrao19 — DISMISSED** [review]

File: rgw/v2/tests/curl/reusable.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    delete_object_status = utils.exec_shell_cmd(command)
+    if delete_object_status is False:
+        raise TestExecError("object deletion failed")
+    log.info(f"object {s3_object_name} deleted")
+    return True
+
+
+def delete_bucket(curl_auth, bucket_name):
+    """
+    delete object using curl
```
Comment (ckulal) [inline rgw/v2/tests/curl/reusable.py:209]: please correct the statement,should be "delete bucket using curl"

File: rgw/v2/tests/curl/reusable.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    ex: curl -X DELETE http://10.0.209.142:80/bkt1/obj1
```
Comment (ckulal) [inline rgw/v2/tests/curl/reusable.py:210]: please change the example as well

File: rgw/v2/tests/curl/test_rgw_using_curl.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info(f"TEST_DATA_PATH: {TEST_DATA_PATH}")
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(description="RGW S3 rgw operations using Curl")
+        parser.add_argument("-c", dest="config", help="RGW S3 operations using AWS")
```
Comment (ckulal) [inline rgw/v2/tests/curl/test_rgw_using_curl.py:118]: please correct the description accordingly

**Comment by ckulal** [comment]
@hmaheswa there are few minor typo changes requested,
except that changes Looks Good

File: rgw/v2/tests/curl/reusable.py (hunk 1 above)
Comment (hmaheswa) [reply rgw/v2/tests/curl/reusable.py]: @ckulal done, please review again!

File: rgw/v2/tests/curl/reusable.py (hunk 2 above)
Comment (hmaheswa) [reply rgw/v2/tests/curl/reusable.py]: @ckulal done, please review again!

File: rgw/v2/tests/curl/test_rgw_using_curl.py (hunk 3 above)
Comment (hmaheswa) [reply rgw/v2/tests/curl/test_rgw_using_curl.py]: @ckulal done, please review again!

**Review by ckulal — APPROVED** [review]

**Pull request merged**